### PR TITLE
Result.ensure/2 and friends

### PR DESCRIPTION
**Pull request by giddie**

### CODE REVIEW

File: lib/ok_then/result.ex [hunk 1]
```diff
@@ -493,6 +493,93 @@ defmodule OkThen.Result do
...
+      iex> nil |> Result.error_ensure(&String.length(&1) == 0)
+      nil
+  """
+  @spec error_ensure(result_input(), (any() -> boolean())) :: result_input()
+  def error_ensure(result, check_function) when is_function(check_function, 1),
+    do: tagged_ensure(result, :error, check_function)
+
+  @doc """
+  If `result` is tagged with the specified `tag` atom, passes the wrapped value into the provided
+  function. If `check_function` returns `true` the `result` is returned unchanged. Otherwise,
```
Comment (tshakah) [inline lib/ok_then/result.ex]: Technically it's just if it returns a truthy value, right?

**Review by tshakah — DISMISSED** [review]
Approved, but if my comment is correct we might want to make that clearer in the docs

File: lib/ok_then/result.ex (hunk 1 above)
Comment (giddie) [reply lib/ok_then/result.ex]: Yes, quite right :+1: Will fix.

**Comment by giddie** [comment]
Rust calls this function `filter` and I think that's a better name. It matches `map`, and essentially is to `Enum.filter` what `map` is to `Enum.map`.

**Review by tshakah — APPROVED** [review]

**Comment by giddie** [comment]
Merged manually :+1:

**Pull request closed without merging**